pkg/controllers/networkpolicy_controller.go

```diff
@@ -0,0 +1,152 @@
+package controllers
+
+import (
+	"context"
+	"strings"
+
+	istiosecurityv1 "istio.io/client-go/pkg/apis/security/v1"
+	netv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	log "github.com/adevinta/go-log-toolkit"
+	ipamv1alpha1 "github.com/adevinta/ingress-allowlisting-controller/pkg/apis/ipam.adevinta.com/v1alpha1"
+	ipamv1alpha1_legacy "github.com/adevinta/ingress-allowlisting-controller/pkg/apis/legacy/v1alpha1"
+	"github.com/adevinta/ingress-allowlisting-controller/pkg/resolvers"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type NetworkPolicyReconciler struct {
+	client.Client
+	Scheme             *runtime.Scheme
+	LegacyGroupVersion string
+	CidrResolver       resolvers.CidrResolver
+}
+
+// +kubebuilder:rbac:groups=networking.k8s.io,resources=networkpolicies,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=ipam.adevinta.com,resources=cidrs,verbs=get;list;watch
+
+func (r *NetworkPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.DefaultLogger.WithContext(ctx).WithField("networkpolicy", req.NamespacedName)
+
+	var networkPolicyMetadata metav1.ObjectMeta
+
+	networkpolicy := netv1.NetworkPolicy{}
+
+	if err := r.Get(ctx, req.NamespacedName, &networkpolicy); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	networkPolicyMetadata = networkpolicy.ObjectMeta
+
+	if !networkPolicyMetadata.DeletionTimestamp.IsZero() { // np being deleted
+		return ctrl.Result{}, nil
+	}
+
+	log.Infof("Networkpolicy %s being reconciled. Creating/updating allowlist...", networkPolicyMetadata.GetName())
+
+	updatedNetworkPolicy, err := r.reconcileNetworkPolicy(ctx, networkpolicy)
+	if err != nil {
+		log.Error(err, "Error creating or updating networkpolicy")
+		return ctrl.Result{}, err
+	}
+
+	networkpolicy = updatedNetworkPolicy
+	if err := r.Client.Update(ctx, &networkpolicy); err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	return ctrl.Result{}, nil
+}
+func (r *NetworkPolicyReconciler) reconcileNetworkPolicy(ctx context.Context, networkPolicy netv1.NetworkPolicy) (netv1.NetworkPolicy, error) {
+	cidrs, err := r.CidrResolver.GetCidrsFromObject(ctx, &networkPolicy)
+	if err == r.CidrResolver.AnnotationNotFoundError() {
+		return networkPolicy, err
```

Comment:
you should maybe emit an error metric or an event here to inform users

Reply:
I think we do
  we generate error in line L54

Reply:
we just need to not notify when annotation is missing